ui/hooks/useAssetDetails.js

```diff
@@ -1,22 +1,37 @@
+import { isEqual } from 'lodash';
 import { useState, useEffect } from 'react';
 import { useSelector, useDispatch } from 'react-redux';
-import { getCollectibles } from '../ducks/metamask/metamask';
+import { getCollectibles, getTokens } from '../ducks/metamask/metamask';
 import { getAssetDetails } from '../helpers/utils/token-util';
 import { hideLoadingIndication, showLoadingIndication } from '../store/actions';
+import { isEqualCaseInsensitive } from '../../shared/modules/string-utils';
 import { usePrevious } from './usePrevious';
+import { useTokenTracker } from './useTokenTracker';
 
 export function useAssetDetails(tokenAddress, userAddress, transactionData) {
   const dispatch = useDispatch();
   // state selectors
   const collectibles = useSelector(getCollectibles);
+  const tokens = useSelector(getTokens, isEqual);
+  const currentToken = tokens.find((token) =>
+    isEqualCaseInsensitive(token.address, tokenAddress),
+  );
 
   // in-hook state
   const [currentAsset, setCurrentAsset] = useState(null);
 
+  let tokensWithBalance;
+  if (currentToken) {
+    // eslint-disable-next-line react-hooks/rules-of-hooks
```

Comment:
thinking of ways to avoid disabling eslint on this line.

what if we did
```
  const { tokensWithBalances } = useTokenTracker(
    currentToken ? [currentToken] : null,
  );
```

with an added null check in `useTokenTracker`? I think this works assuming `state.metamask.tokens` is always an array even when it's loading. I'm assuming this is also true based on the logic from line 16 above

Reply:
I think we should be able to achieve the same effect by just passing `useTokenTracker` an empty array? (and then the `useTokenTracker` hook would not have to be modified

```
  const { tokensWithBalances } = useTokenTracker(
    currentToken ? [currentToken] : [],
  );
 ```
 
 I agree that we should not disable eslint rules for hooks

Reply:
Thank you for suggestion, I updated my PR